app.ts

```diff
@@ -0,0 +1,218 @@
+let lastNumber = 0;
+let backend: string = "http://localhost:2050";
+
+window.onload = () => {
+	createTable();
+	displayRecords();
+	$('#btnSearch').on("click", async (event) => {
+		event.preventDefault();
+		let inputValue = $('#searchInput').val() as number;
+		await updateRecordsAndResize(inputValue); // calls to calculate the range once button is clicked
+	});
+	$('#closeModalBtn').on("click", () => {
+		$('.content').empty()
+		$('.modal').css('display', 'none') // closes modal
+	});
+	$('.arrow-right').on('click', () => {
+		rightArrow();
+	});
+	$('.arrow-left').on('click', () => {
+		leftArrow();
+	});
+	$('#searchInput').on('keydown', (event) => {
+		if (event.key === 'e' || event.key === 'E') {
+			event.preventDefault();
+		};
+	});
+	$('#searchInput').on('input', (event) => {
+		const inputValue = $('#searchInput').val() as string;
+		if (inputValue.includes('.')) {
+			$('#searchInput').val(inputValue.replace('.', ' '));
+		};
+	});
+};
+
+async function fetchRecordCount(): Promise<number> {
+	try {
+		const response = await fetch(`${backend}/recordCount`);
+		if (!response.ok) {
+			throw new Error('Failed to fetch record count');
+		}
+		return response.json()
+	} catch (error) {
+		throw new Error('Error fetching the record count')
+	};
+};
+
+async function fetchColumns(): Promise<string[]> {
+	try {
+		const response = await fetch(`${backend}/columns`);
+		if (!response.ok) {
+			throw new Error('Failed to fetch columns');
+		}
+		const jsonText = await response.text();
+		const columns: string[] = await JSON.parse(jsonText);
+		return columns
+	} catch (error) {
+		throw new Error('Error fetching columns')
+	};
+};
+
+async function createTable() {
+	try {
+		const columns = await fetchColumns();
+		for (const col of columns) {
+			$(".head").append(`<th>${col}</th>`);
+		};
+	} catch (error) {
+		throw new Error('Error creating table');
+	};
+};
+
+function adjustRowsByScreenHeight() {
+	const screenHeight = window.innerHeight;
+	const availableHeight = screenHeight - 105; // subtracts the space used from the screeen
+	let rowHeight = 35;
+	if (availableHeight <= 0) {
+		return 0;
+	} else {
+		let maxRows = Math.floor(availableHeight / rowHeight);
+		return maxRows;
+	};
+};
+
+$(window).on('resize', async () => {
+	let resizeTimeout: number;
+	resizeTimeout = setTimeout(async () => {
+		$('#loader').show()
+		await displayRecords()
+		let inputValue = $('#searchInput').val(); // priotizes the search input value if available
+		if (inputValue !== '') {
+			await updateRecordsAndResize(Number(inputValue))
+		}
+		$('#loader').hide();
+	}, 500);
+});
+
+async function fetchRecords(from: number, to: number): Promise<any[]> {
+	try {
+		const response = await fetch(`${backend}/records?from=${from}&to=${to}`);
+		if (!response.ok) {
+			throw new Error("Sorry, there's a problem with the network");
+		}
+		return response.json();
+	} catch (error) {
+		throw new Error('Error fetching records from server');
+	};
```

Comment:
No need for the async anymore, and can remove the try and catch
```suggestion
	return fetch(`${backend}/records?from=${from}&to=${to}`)
		.then(res => {
			if (!res.ok) {
				throw "Sorry, there's a problem with the network";
			}
			return res.json();
		}).catch(err => {
			throw 'Error fetching records from server ' + err;
		});
```